### PR TITLE
[tx] Introduce optimization step metrics dataclass

### PR DESCRIPTION
**Pull request by pcmoritz**
This is in preparation for merging https://github.com/NovaSky-AI/SkyRL/pull/1008 and to make it easier to introduce metrics.

---

<a href="https://app.devin.ai/review/novasky-ai/skyrl/pull/1191" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: skyrl-tx/tx/tinker/backends/jax.py [hunk 1]
```diff
@@ -738,8 +756,7 @@ def optim_step(self, model_id: str, request_data: types.OptimStepInput) -> types
 
         # Check if we have any gradients accumulated (count > 0)
         if self.accumulated_grads.counts[adapter_index] == 0:
-            logger.warning(f"No accumulated gradients for model {model_id}, skipping optimizer step")
-            return types.OptimStepOutput(metrics={"skyrl.ai/learning_rate": learning_rate})
+            logger.warning(f"No accumulated gradients for model {model_id}; applying step with zero gradients")
```
[the remark this answers is not in the document]
Comment (pcmoritz) [reply skyrl-tx/tx/tinker/backends/jax.py:759]: I don't think the semantics of an empty gradient step are well specified, but applying an empty optimizer step seems like a reasonable thing to do and is also in line with reporting a zero gradient norm. This case shouldn't be common, and handling everything uniformly (zero gradient norm, apply optimizer step) seems like good semantics to me.
